File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.py
class Solution(object):
    def minimumDeletions(self, s):
        n = len(s)
        cnt_a = [0] * (n + 1)
        cnt_b = [0] * (n + 1)
    
        for i in range(1, n + 1):
            cnt_a[i] = cnt_a[i - 1] + (1 if s[i - 1] == 'a' else 0)
            cnt_b[i] = cnt_b[i - 1] + (1 if s[i - 1] == 'b' else 0)
    
        min_dels = float('inf')
    
        for i in range(n + 1):
            dels = cnt_b[i] + (cnt_a[n] - cnt_a[i])
            min_dels = min(min_dels, dels)
    
        return min_dels

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.c
int minimumDeletions(char *s) {
    int n = strlen(s);
    int cnt_a[n + 1];
    int cnt_b[n + 1];
    
    cnt_a[0] = 0;
    cnt_b[0] = 0;

    for (int i = 1; i <= n; i++) {
        cnt_a[i] = cnt_a[i - 1] + (s[i - 1] == 'a' ? 1 : 0);
        cnt_b[i] = cnt_b[i - 1] + (s[i - 1] == 'b' ? 1 : 0);
    }

    int min_dels = INT_MAX;

    for (int i = 0; i <= n; i++) {
        int dels = cnt_b[i] + (cnt_a[n] - cnt_a[i]);
        if (dels < min_dels) {
            min_dels = dels;
        }
    }

    return min_dels;
}

